common-api/src/main/scala/pl/touk/nussknacker/engine/api/MetaData.scala

```diff
@@ -6,9 +6,14 @@ import io.circe.generic.extras.semiauto.{deriveConfiguredDecoder, deriveConfigur
 import io.circe.{Decoder, Encoder, HCursor}
 import pl.touk.nussknacker.engine.api.process.ProcessName
 import pl.touk.nussknacker.engine.api.CirceUtil._
+import sttp.tapir.Schema
 
 @JsonCodec case class LayoutData(x: Long, y: Long)
 
+object LayoutData {
+  implicit val schema: Schema[LayoutData] = Schema.derived
```

Comment:
IMO tapir schema should be defined close to tapir endpoint definitions. We should also ask ourselves if circe coders should be defined here too

Reply:
I agree in a case when this is DTO used only for the sake of given endpoint. But what if structure is used a substructure in multiple endpoints (like `ScenarioGraph` class in my case)? Then according to your advice it would cause code repetition.

Reply:
if it's shared by several tapir services, the schema and codecs should be defined in tapir's endpoints package (see eg. `TapirCodecs`).
Adding this dependency to Nu API module is a rather bad idea.

Reply:
Okay, thanks for pointing this out!

Reply:
I relaxed the schema since I do not think it is worth the price. Instead I could expand provided example. Of course it might be marked as TODO for the brave ones.